crypto/mem_sec.c

```diff
@@ -25,10 +25,11 @@
 # include <sys/param.h>
 # include <sys/stat.h>
 # include <fcntl.h>
+# include "internal/threads.h"
 #endif
 
-#define LOCK()      CRYPTO_w_lock(CRYPTO_LOCK_MALLOC)
-#define UNLOCK()    CRYPTO_w_unlock(CRYPTO_LOCK_MALLOC)
+#define LOCK()      CRYPTO_THREAD_write_lock(sec_malloc_lock)
```

Comment:
now that these are portable, replace the LOCK/UNLOCK with their values in all occurrences below.